waku/incentivization/rpc_codec.nim

```diff
@@ -0,0 +1,81 @@
+import
+  std/options
+import
+  ../common/protobuf,
+  ../waku_core,
+  ./rpc
+
+
+proc encode*(rpc: DummyRequest): ProtoBuffer = 
+  var pb = initProtoBuffer()
+
+  pb.write3(1, rpc.requestId)
+
+  if rpc.eligibilityProof.isSome:
+    let ep = rpc.eligibilityProof.get()
+    pb.write3(2, uint32(ord(ep.proofType)))
+    pb.write3(3, ep.proof)
```

Comment:
Since `EligibilityProof` is a separate protobuf message, why not reset the tag numbers here (they cannot clash with the encapsulating Request's).